### ISSUE TITLE
Password Policy

### ISSUE THREAD

**Issue by Moschn** [amiv-eth/amivapi#234]
There should be a password policy. We already use a good hash function for passwords but a password with length 1 is currently allowed and easily broken. 

I propose just a minimum length of 7 for passwords containing only letters. This would lead to a searching space of 24^7 and with a required time of around 100ms per hash it would take 14.5 years.

In order to prevent password list attacks, a more complicated policy with alphanumeric and sepcial character requirements could also be useful.

**Comment by cburchert** [comment]
I don't like password policies. I basically use a top 10 common password on
every site I don't care about, also to prevent them from leaking any of my
real passwords. I think many people might feel this way about AMIV too, as
they only login to register for events or look at studydocs. We do not
store valueable data people care about for most students (I know the amount
people care about their data differs a lot from person to person).

It makes sense to have a policy for accounts with extented permissions,
e.g., being able to look at the data of other students. However I would
prefer to achieve this with other means, such as having a text on the
password setting form which encourages people with more privileges to use
strong passwords and password managers. Furthermore I think there is
basically no overlap of the users that don't login over LDAP and the users
that have privileges.

2018-02-10 16:58 GMT+01:00 Moritz Schneider <notifications@github.com>:

**Comment by cburchert** [comment]
As and addendum: People find ways around most password rules anyway. If I
don't care about a website and it wants a strong password I will basically
always add the required characters in the simplest form, e.g. use 123456aA!
if they require me to use numbers, lower case letters, upper case letters
and also a symbol.

2018-02-10 20:27 GMT+01:00 Conrad Burchert <conrad.burchert@gmail.com>:

**Comment by temparus** [comment]
Even if you use the required characters in the simplest form will provide better security that a one letter password. 
In my opinion, it is negligent to leave it as it is now.

**Comment by hermannsblum** [comment]
I think the only valid password policy is to force the password to a minimum length of 40 characters, which has to begin with the prefix `what I hate most about vis `

**Comment by NotSpecial** [comment]
Furthermore, the length should be divisible by four, and the numerical values of each part summed together (mod 26) as a string should result in 'amiv'

**Comment by NotSpecial** [comment]
More reasonably: I want to stress out what @cburchert said: Most people will *not* be setting their password, since the LDAP login just works and requires no login at all.

So I guess the only people setting a password are external members *not* at ETH (and there are not very many, I think) and people who are mindful about security and want to choose a separate password for amiv, and I guess those people do not require a password policy.

I think @Moschn 's proposal is reasonable: A minimum length seems reasonable, but I would not put too much effort in other policies.
